contrib/network/connection_balance/dlb/test/config_test.cc

```diff
@@ -102,10 +147,10 @@ TEST_F(DlbConnectionBalanceFactoryTest, TooManyThreads) {
 
   envoy::extensions::network::connection_balance::dlb::v3alpha::Dlb dlb;
   makeDlbConnectionBalanceConfig(typed_config, dlb);
-
-  EXPECT_THAT_THROWS_MESSAGE(
-      factory.createConnectionBalancerFromProto(typed_config, context), EnvoyException,
-      HasSubstr("Dlb connection balanncer only supports up to 32 worker threads"));
+  EXPECT_LOG_CONTAINS("warn", "Dlb connection balanncer only supports up to 32 worker threads",
+                      factory.createConnectionBalancerFromProto(typed_config, context));
+  EXPECT_EQ(std::make_shared<Network::NopConnectionBalancerImpl>(),
```

Comment:
Might be missing something, is the default behavior of `None` fallback to create the NopConnectionBalancerImpl? I would have expected it to just throw as it was doing prior. Seems related to discussion here https://github.com/envoyproxy/envoy/pull/27985#discussion_r1232180361